### ISSUE TITLE
Failed to converge when training the model.

### ISSUE THREAD

**Issue by lconan** [BIMSBbioinfo/ikarus#17]
Hi, thanks for this great tool.
I managed to get the gmt file with public datasets. But I ran into the error when I try to train the model:
```
/home/sxykdx/miniconda3/envs/ikarus/lib/python3.9/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
```
So, I try to edit here:
https://github.com/BIMSBbioinfo/ikarus/blob/0a4baf40030f998548c18beb93cb06dd9a774420/ikarus/classifier.py#L533

Changed default from 1000 to 2000,4000,10000,40000; still got the same error message. I'm not sure how to fix this. Maybe, I should subsample the data? Thx

**Comment by lconan** [comment]
I should try assign n_iter in classifier.Ikarus()

**Comment by lconan** [comment]
Still can't converge, and the time cost didn't change much, as n_iter increased from 50 to 200, 1000,4000 and 10k

**Comment by dohmjan** [comment]
Hi! Could you try to increase `max_iter` [here](https://github.com/BIMSBbioinfo/ikarus/blob/master/ikarus/classifier.py#L20)? I think its default value is 100.

The `max_iter` and `n_iter` you changed above refer to later steps, the cnv part and the label propagation, respectively.

**Comment by lconan** [comment]
> Hi! Could you try to increase `max_iter` [here](https://github.com/BIMSBbioinfo/ikarus/blob/master/ikarus/classifier.py#L20)? I think its default value is 100.
> 
> The `max_iter` and `n_iter` you changed above refer to later steps, the cnv part and the label propagation, respectively.

thanks, it'll do the job without subsampling the training data.
The thing is that I downloaded 3 available datasets of AML sample, 2 as training data, 1 as validation set. I tried 2 cominations. The first classified most of the cells as Tumor, the 2nd classified most of the cells as Normal. 
I also tried to change the `top_x` parameter of `gene_list.create_all()` to try different ratios of normal genes signature to tumor genes signature. It might be good to have snv or cnv data, which I'm not able to get access to. Do you mind to give me some suggestions? Thanks

**Comment by melonheader** [comment]
Closing the issue now due to inactivity. Feel free to re-open if the problem persists.
